modules/common/development/scripts/perf_test_icicle_kit.nix

```diff
@@ -0,0 +1,20 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{pkgs, ...}:
```

Comment:
Don't put pkgs here if you are using callPackage mechanism.

The callPackage mechansim works so, you can put every package name here into this argument type attribute set, which seems to be just `{writeScriptBin}` at this point.

I think generally in all these files it would be better idea to use writeShellApplication instead of writeScriptBin